File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area = 0;
        for(int l = 0, r = height.size() - 1; l < r;)
        {
            max_area = max(max_area, ( (r - l) * min(height[l], height[r])));
            height[l] < height[r] ? l++ : r--;
        }
        return max_area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int l = 0, r = heights.Length - 1;
        int maxArea = 0;
        while (l < r)
        {
            int water_stored = Math.Min(heights[l],heights[r]) * (r - l);
            maxArea = Math.Max(water_stored, maxArea);
            if (heights[l] < heights[r])
                l++;
            else
                r--;
        }
        return maxArea;
    }
}
